scheduler/scheduler-examples/src/main/java/org/ow2/proactive/scheduler/examples/FTPConnector.java

```diff
@@ -0,0 +1,374 @@
+            }
+        }
+        if (args.containsKey("ftpExtractArchive")) {
+            ftpExtractArchive = Boolean.parseBoolean(args.get("ftpExtractArchive").toString());
+        }
+        if (args.containsKey("ftpLocalFilePath")) {
+            ftpLocalFilePath = args.get("ftpLocalFilePath") == null ? "" : args.get("ftpLocalFilePath").toString();
+        }
+        if (args.containsKey("ftpRemoteFilePath")) {
+            ftpRemoteFilePath = args.get("ftpRemoteFilePath").toString();
+        }
+        if (args.containsKey("ftpMode")) {
+            ftpMode = args.get("ftpMode").toString();
+        }
+        ftpUsername = getThirdPartyCredential("FTP_USERNAME");
+        ftpPassword = getThirdPartyCredential("FTP_PASSWORD");
+        if (ftpUsername == null || ftpPassword == null) {
+            throw new IllegalArgumentException("You first need to add your ftp username and password (FTP_USERNAME, FTP_PASSWORD) to the third party credentials");
+        }
+    }
+
+    /**
+     * @see JavaExecutable#execute(TaskResult[])
+     */
+    @Override
+    public Serializable execute(TaskResult... results) throws IOException, IllegalArgumentException {
+
+        FTPClient ftpClient = new FTPClient();
+        // connect and login to the server
+        ftpClient.connect(ftpHostname, ftpPort);
+        int reply = ftpClient.getReplyCode();
+        if (!FTPReply.isPositiveCompletion(reply)) {
+            ftpClient.disconnect();
+            throw new IOException("Exception in connecting to FTP Server");
+        }
+        ftpClient.login(ftpUsername, ftpPassword);
+
+        // use local passive mode to pass firewall
+        ftpClient.enterLocalPassiveMode();
+
+        getOut().println("Connected");
+
+        //FTP mode is GET
+        if (ftpMode.equals("GET")) {
+            getOut().println("Importing file(s) from " + ftpRemoteFilePath + " to " + ftpLocalFilePath);
+            FTPFile[] ftpFile = ftpClient.listFiles(ftpRemoteFilePath);
+            if (ftpFile.length == 0)
+                throw new IllegalArgumentException(ftpRemoteFilePath + " not found. Please, enter a valid path.");
+
+            // If it is a single file:
+            if (ftpFile.length == 1 && ftpRemoteFilePath.contains(ftpFile[0].getName())) {
+                String saveFilePath = Paths.get(ftpLocalFilePath, ftpFile[0].getName()).toString();
+                boolean download = downloadSingleFile(ftpClient, ftpRemoteFilePath, saveFilePath);
+                if (download) {
+                    getOut().println("Succeed to download the file " + ftpRemoteFilePath + " to " + ftpLocalFilePath);
+                } else {
+                    getOut().println("COULD NOT download the file: " + ftpRemoteFilePath);
+                }
+
+                // If the file is a zip, and ftpExtractArchive is set to true
+                if (ftpExtractArchive && ftpRemoteFilePath.endsWith(".zip")) {
+                    ZipFile zipFile = null;
+                    try {
```

Comment:
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') Change this "try" to a try-with-resources. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS2093)